src/runtime/virtcontainers/qemu_arm64.go

```diff
@@ -107,3 +108,14 @@ func (q *qemuArm64) setIgnoreSharedMemoryMigrationCaps(_ context.Context, _ *gov
 func (q *qemuArm64) appendIOMMU(devices []govmmQemu.Device) ([]govmmQemu.Device, error) {
 	return devices, fmt.Errorf("Arm64 architecture does not support vIOMMU")
 }
+
+func (q *qemuArm64) append9PVolume(devices []govmmQemu.Device, volume types.Volume) ([]govmmQemu.Device, error) {
```

Comment:
I'm assuming this is a temporary fix until the arm64 version of qemu is updated. However, this function is essentially a copy of `qemuArchBase.append9PVolume()`. To protect against that changing (and this version not changing, it might be better to do something like this:

```diff
diff --git a/src/runtime/virtcontainers/qemu_arch_base.go b/src/runtime/virtcontainers/qemu_arch_base.go
index e2ab2360..efd438d6 100644
--- a/src/runtime/virtcontainers/qemu_arch_base.go
+++ b/src/runtime/virtcontainers/qemu_arch_base.go
@@ -464,16 +464,20 @@ func generic9PVolume(volume types.Volume, nestedRun bool) govmmQemu.FSDevice {
        }
 }

-func (q *qemuArchBase) append9PVolume(devices []govmmQemu.Device, volume types.Volume) ([]govmmQemu.Device, error) {
+func genericAppend9PVolume(devices []govmmQemu.Device, volume types.Volume, nestedRun bool) ([]govmmQemu.Device, error) {
        if volume.MountTag == "" || volume.HostPath == "" {
                return devices, nil
        }

-       d := generic9PVolume(volume, q.nestedRun)
+       d := generic9PVolume(volume, nestedRun)
        devices = append(devices, d)
        return devices, nil
 }

+func (q *qemuArchBase) append9PVolume(devices []govmmQemu.Device, volume types.Volume) ([]govmmQemu.Device, error) {
+       return genericAppend9PVolume(devices, volume, q.nestedRun)
+}
+
 func (q *qemuArchBase) appendSocket(devices []govmmQemu.Device, socket types.Socket) []govmmQemu.Device {
        devID := socket.ID
        if len(devID) > maxDevIDSize {
diff --git a/src/runtime/virtcontainers/qemu_arm64.go b/src/runtime/virtcontainers/qemu_arm64.go
index 93eb79d1..5eb0cdb5 100644
--- a/src/runtime/virtcontainers/qemu_arm64.go
+++ b/src/runtime/virtcontainers/qemu_arm64.go
@@ -110,12 +110,13 @@ func (q *qemuArm64) appendIOMMU(devices []govmmQemu.Device) ([]govmmQemu.Device,
 }

 func (q *qemuArm64) append9PVolume(devices []govmmQemu.Device, volume types.Volume) ([]govmmQemu.Device, error) {
-        if volume.MountTag == "" || volume.HostPath == "" {
-                return devices, nil
+       d, err := genericAppend9PVolume(devices, volume, q.nestedRun)
+       if err != nil {
+               return nil, err
        }

-        d := generic9PVolume(volume, q.nestedRun)
+       // arm64 version of qemu is too old to support the multidev parameter
        d.Multidev = ""
-        devices = append(devices, d)
-        return devices, nil
+
+       return d, nil
 }
```

Reply:
thanks @jodh-intel, I tried this, but will fail at 'd.Multidev = "" ' as this "d" is in type of govmmQemu.Device and can't access "Multidev", after address this problem, I find that the "genericAppend9PVolume" can be just a encapsulation of "generic9PVolume"